src/GeositeFramework/js/util/ajax.js

```diff
@@ -11,16 +11,27 @@ define(['esri/request'],
         }
 
         // Fetch data from `url` and store the result in `cache` (memoized).
-        var fetch = function(url) {
+        var fetch = function(url, format, postprocess) {
+            var content = null,
+                format = format || 'json';
+
+            if (format === 'json') {
+                content = {f: 'json'};
+            }
+
             if (typeof promises[url] === 'undefined') {
                 promises[url] = request({
                     url: url,
-                    content: {f: 'json'},
-                    handleAs: 'json',
+                    content: content,
+                    handleAs: format,
                     callbackParamName: 'callback',
-                    timeout: 7000
+                    timeout: 20000
                 }).then(function(data) {
-                    cache[url] = data;
+                    if (postprocess && typeof postprocess === 'function') {
+                        cache[url] = postprocess(data);
```

Comment:
I would vote to remove `postprocess` and let the WMS service handle this transformation and maybe manage its own cache. Since this cache is global I think only raw data should be preserved.